biz.aQute.bndlib/src/aQute/bnd/service/reporter/BundleEntryPlugin.java

```diff
@@ -0,0 +1,41 @@
+package aQute.bnd.service.reporter;
+
+import java.util.Locale;
+
+import org.osgi.annotation.versioning.ProviderType;
+
+import aQute.bnd.osgi.Jar;
+import aQute.bnd.osgi.Processor;
+
+/**
+ * This plugin extracts a piece of information from a Jar and converts it into a
+ * DTO representation. Its result will contributes to the formation of a Jar
+ * metadata report.
+ */
+@ProviderType
+public interface BundleEntryPlugin {
+
+	/**
+	 * Extracts a piece of information from the Jar in arguments.
+	 * <p>
+	 * If the Jar contains localized data, it will be extracted for the
+	 * specified locale or a less specific if not found. The syntax of the
+	 * locale must be as specified in {@link Locale}, an empty {@code String}
+	 * means {@code unlocalized}.
+	 * 
+	 * @param jar the Jar to inspect, must not be {@code null}
+	 * @param locale the {@code String} representation of a {@code Locale}, must
+	 *            not be {@code null}
+	 * @param reporter used to report error, must not be {@code null}
+	 * @return a DTO representation or {@code null} if no data is available
+	 */
+	public Object extract(final Jar jar, final String locale, final Processor reporter) throws Exception;
```

Comment:
The use of Object as the return type in these plugins is a bit disconcerting. It means that only reflective access to the DTOs is possible. Why would we not want to define the DTO types and then return them here?

Reply:
The consumer just uses these plugins to collect DTO and serialization.